src/compas_timber/model/model.py

```diff
@@ -102,58 +106,44 @@ def beam_by_guid(self, guid):
 
         Returns
         -------
-        :class:`~compas_timber.elements.Beam`
-            The beam with the specified GUID.
+        :class:`~compas_timber.elements.Element`
+            The element with the specified GUID.
 
         """
         return self._guid_element[guid]
 
-    def add_beam(self, beam):
+    def add_element(self, element):
         # type: (Beam) -> None
         """Adds a Beam to this model.
 
         Parameters
         ----------
-        beam : :class:`~compas_timber.elements.Beam`
-            The beam to add to the model.
-
-        """
-        _ = self.add_element(beam)
-        self._beams.append(beam)
-
-    def add_wall(self, wall):
-        # type: (Wall) -> None
-        """Adds a Wall to this model.
-
-        Parameters
-        ----------
-        wall : :class:`~compas_timber.elements.Wall`
-            The wall to add to the model.
+        element : :class:`~compas_timber.elements.Element`
+            The element to add to the model.
 
         """
-        _ = self.add_element(wall)
-        self._walls.append(wall)
+        _ = super(TimberModel, self).add_element(element)
 
-    def add_joint(self, joint, beams):
-        # type: (Joint, tuple[Beam]) -> None
+    def add_interaction(self, interaction, elements):
+        # type: (Joint, tuple[Interaction]) -> None
         """Add a joint object to the model.
 
         Parameters
         ----------
-        joint : :class:`~compas_timber.connections.joint`
+        interaction : :class:`~compas_timber.connections.joint`
             An instance of a Joint class.
 
-        beams : tuple(:class:`~compas_timber.elements.Beam`)
-            The two beams that should be joined.
+        beams : tuple(:class:`~compas_timber.elements.Element`)
+            The two elements that should be joined.
 
         """
-        if len(beams) != 2:
-            raise ValueError("Expected 2 parts. Got instead: {}".format(len(beams)))
-        a, b = beams
-        _ = self.add_interaction(a, b, interaction=joint)
-        self._joints.append(joint)
+        if len(elements) != 2:
+            raise ValueError("Expected 2 parts. Got instead: {}".format(len(elements)))
+        a, b = elements
+        print(a, b)
+        _ = super(TimberModel, self).add_interaction(a, b, interaction=interaction)
 
-    def remove_joint(self, joint):
+    def remove_interaction(self, joint):
```

Comment:
Good question. I was thinking about this in terms of `Joint` being a specific type of the generic type `Interaction`. However, without any other types of interaction we could leave it as `remove_joint`

Reply:
keep in mind that since `TimberModel` inherits from `Model`, the `Model` interface is fully available on `TimberModel` objects. We could totally have `remove_joint`/`add_joint` live alongside `remove_interaction`/`add_interaction`.

Any specific operation to interactions of type `Joint` can be handled in `add_joint` whereas other kinds of interactions that are not timber model specific can still be added using `add_interaction`.